Sources/Testing/ExitTests/WaitFor.swift

```diff
@@ -43,6 +43,44 @@ private func _blockAndWait(for pid: pid_t) throws -> ExitCondition {
   }
 }
 
+#if os(Linux)
+/// A mapping of awaited child PIDs to their corresponding Swift continuations.
+private let _childProcessContinuations = Locked<[pid_t: CheckedContinuation<Void, Never>]>()
+
+/// The implementation of `_createWaitThread()`, run only once.
+private let _createWaitThreadImpl: Void = {
+  // Create the thread. We immediately detach it upon success to allow the
+  // system to reclaim its resources when done.
+
+  var thread = pthread_t()
+  _ = pthread_create(
+    &thread,
+    nil,
+    { _ in
+      // Run an infinite loop that waits for child processes to terminate and
+      // captures their exit statuses.
+      while true {
+        var siginfo = siginfo_t()
+        if 0 == waitid(P_ALL, 0, &siginfo, WEXITED | WNOWAIT) {
+          let pid = swt_siginfo_t_si_pid(&siginfo)
+          let continuation = _childProcessContinuations.withLock { childProcessContinuations in
```

Comment:
There's a race condition here if `waitid(P_ALL)` fires for a PID before we've added it to the dictionary.